lib/ast-utils.js

```diff
@@ -84,11 +84,10 @@ function isES5Constructor(node) {
  * @returns {Node|null} A found function node.
  */
 function getUpperFunction(node) {
-    while (node) {
-        if (anyFunctionPattern.test(node.type)) {
-            return node;
+    for (let currentNode = node; currentNode; currentNode = currentNode.parent) {
```

Comment:
This is completely personal preference, and can be ignored for this PR, but I don't like using `for` loops for something other then counters. It's unexpected, and makes it much harder to read the code.